Assets/Scripts/InputManager.cs

```diff
@@ -52,8 +59,80 @@ private void Update()
 
 //a struct to organize each players custom inputs. 
 [System.Serializable]
-public struct PlayerControl
+public class PlayerControl
 {
-    //these are not final ofcourse.
-    public KeyCode Shoot, Dash, Ability1, Ability2, Ability3;
+    const float moveValue = 0.5f;
+
+    //player inputs to listen to. //todo support gamepads
+    public KeyCode MoveUp, MoveDown, MoveLeft, MoveRight, Shoot, Dash, Ability1, Ability2, Ability3;
+
+    //check if player is moving.
+    public bool Moving {
+        get
+        {
+            return (Input.GetKey(MoveUp) || Input.GetKey(MoveDown) ||
+                   Input.GetKey(MoveLeft) || Input.GetKey(MoveRight));
+        }
+    }
+
+    //check if player just stopped.
+    public bool Stopped
+    {
+        get
+        {
+            return (Input.GetKeyUp(MoveUp) || Input.GetKeyUp(MoveDown) ||
+                   Input.GetKeyUp(MoveLeft) || Input.GetKeyUp(MoveRight));
```

Comment:
Won't this incorrectly return true when, for example, the player is holding down MoveUp and MoveRight, and then when he releases MoveRight but is still holding MoveUp?

Reply:
Nope. As I pass a move value to the player.. and it's based off what buttons are down.
The GetKeyUp stuff is just so i can call move again .. but with move values of 0.